### ISSUE TITLE
Postgres URL support

### ISSUE THREAD

**Issue by jakajancar** [bikeshedder/deadpool#58]
`tokio_postgres::Config` supports URLs, but as far as I can see deadpool doesn't, meaning one has to drop using URLs (and some systems, such as Heroku, provide URLs).

**Comment by bikeshedder** [comment]
This should already be possible. `tokio_postgres::Config` which is re-exported by `deadpool_postgres::Config` implements the `FromStr` trait. Therefore you can just write...

```rust
let url = "postgresql://user@localhost";
let config = Config::from_str(url)?;
```

Also check out the documentation of [tokio_postgres::config::Config](https://docs.rs/tokio-postgres/0.5.5/tokio_postgres/config/struct.Config.html) yourself.

Hope that helps. :grinning:

**Comment by jakajancar** [comment]
Looking at the code I don't think `deadpool_postgres::config::Config` is a reexport of `tokio_postgres::Config` but a separate struct. Which is then a problem because although `tokio_postgres::config::Config` supports strings, you cannot convert it into `deadpool_postgres::config::Config` without piping every property. But maybe I'm seeing something wrong since I'm new to Rust :) Ultimately, it's why I swiched to bb8: because it allows using `tokio_postgres::config::Config` directly.

**Comment by bikeshedder** [comment]
My fault. I mixed that up. I shouldn't be answering issues middle in the night. :dizzy_face: 

`deadpool_postgres::Config` is just a builder which can be used to create`tokio_postgres::Config` instances. It is entirely optional and just makes using the `config` crate possible.

As a matter of fact you can just create a `tokio_postgres::Config` and use that when creating the pool:

```rust
let url = "postgresql://user@localhost";
let config = tokio_postgres::Config::from_str(url)?;
let manager = deadpool_postgres::Manager::new(config, tokio_postgres::NoTls);
let pool = deadpool_postgres::Pool::new(config, 16);
```

This is actually also in the README of the crate but that version hasn't been published yet:

https://github.com/bikeshedder/deadpool/tree/master/postgres#example-using-an-existing-tokio_postgresconfig-object

**Comment by bikeshedder** [comment]
I just released deadpool-postgres 0.5.6 which includes the new recycling method proposed by you in #57 and also has the correct README on crates.io showing the usage without using the `deadpool-postgres` specific `Config` structure:

https://crates.io/crates/deadpool-postgres/0.5.6

I should've made that release a long time ago. I'm sorry for the confusion this has caused.
